lib/tasks/report.rake

```diff
@@ -40,4 +40,17 @@ namespace :report do
     end
     Reports::BrexitSubscribersReport.call(args[:date])
   end
+
+  desc "Outputs a CSV of information for each subscriber list for a past date, format: 'yyyy-mm-dd'"
+  task :csv_subscriber_lists, [:date] => :environment do |_t, args|
+    date = Time.zone.parse(args[:date])
+
+    if date.blank?
+      puts 'Invalid date. Please use "yyyy-mm-dd"'
```

Comment:
While I agree with the principle of `puts`ing stuff at the task level, I think it would be simpler for the report to do the validation - it parses the time anyway - and throw an error instead of `puts`ing - the task description says it all!

Reply:
Sure! Have converted these to raises and shifted them inside the report class.